src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -93,6 +95,9 @@ public ProvenBlockHeader(PosBlock block)
             this.signature = block.BlockSignature;
             this.coinstake = block.Transactions[1];
             this.merkleProof = new MerkleBlock(block, new[] { this.coinstake.GetHash() }).PartialMerkleTree;
+
+            BlockStake blockStake = BlockStake.Load(block);
+            this.StakeModifierV2 = blockStake.StakeModifierV2;
```

Comment:
I dont think you can just do that unfortunately stake modifier needs to be created based on the previous SMV2

Reply:
this is old code, don't review this, we need to talk (new code is uncommitted because requires design decision)

Reply:
So is this still valid?